### ISSUE TITLE
How to write your differentable-rasterizer, likes our GS GREAT-MASTERS.  (step-1 cuda<HERE>;  step-2 forward and backward/gradients<TODO>)

### ISSUE THREAD

**Issue by yuedajiong** [graphdeco-inria/gaussian-splatting#421]
[mycuda.cu.txt](https://github.com/graphdeco-inria/gaussian-splatting/files/13235487/mycuda.cu.txt)  
TODO(forward and backward)

technical keywords:  
   1. cpp-compiling and python-cpp interface
       torch.utils.cpp_extension.load(my rewrite)   OR   setup.py for pip install (gauss-splat official)
       PYBIND11_MODULE
   2. cuda-related coding
      torch.autograd.Function for our backward(),  and c++ calling

      <<<,>>>    //preprocessCUDA<NUM_CHANNELS> <<<(P + 255) / 256, 256 >>> (...)

      __global__
     __device__
     
     __shared__
      
    auto block = cg::this_thread_block();

    cooperative_groups

reference:
1. mesh, point-cloud,...   @pytorch3d#meta
2. mesh                          @nvdiffrast#nvidia
3. gauss-point                @gaussian-splatting
4. ****                             @StoneBird-Project4-CUDA-Rasterizer
